baselines/fedstar/fedstar/client.py

```diff
@@ -0,0 +1,154 @@
+            batch_size=batch_size,
+            l_per=l_per,
+            u_per=u_per,
+            fedstar=fedstar,
+            class_distribute=class_distribute,
+            mean_class_distribution=mean_class_distribution,
+            balance_dataset=balance_dataset,
+            seed=seed,
+        )
+        self.num_examples_train = (
+            self.num_batches * batch_size if self.train_labelled else 0
+        )
+        # Local Variables Initialize
+        self.local_train_round = 0
+        self.local_evaluate_round = 0
+        self.weights = Network.get_init_weights(num_classes=self.num_classes)
+        self.history: Dict[str, List[float]] = {"loss": [], "accuracy": []}
+
+        if self.fedstar:
+            self.model = PslNetwork(
+                num_classes=self.num_classes, aux_loss_weight=self.aux_loss_weight
+            )
+            self.model.compile(optimizer=tf.keras.optimizers.Adam(float(learning_rate)))
+        else:
+            self.model = Network(num_classes=self.num_classes).get_network()
+            self.model.compile(
+                optimizer=tf.keras.optimizers.Adam(float(learning_rate)),
+                loss=tf.keras.losses.SparseCategoricalCrossentropy(
+                    from_logits=True, name="loss"
+                ),
+                metrics=[tf.keras.metrics.SparseCategoricalAccuracy(name="accuracy")],
+            )  # type: ignore
+
+        # tf.keras.backend.clear_session()
+
+    def get_parameters(self, config):
+        """Return the current client model weights."""
+        return self.weights
+
+    # pylint: disable=no-value-for-parameter
+    def fit(self, parameters, config):
+        """Update the current client model weights."""
+        self.local_train_round += 1
+        # Run Training Proccess
+        if self.fedstar:
+            self.model.set_weights(parameters)
+            history = self.model.fit(
+                (self.train_labelled, self.train_unlabelled),
+                num_batches=self.num_batches,
+                epochs=int(config["epochs"]),
+                c_round=int(config["c_round"]),
+                rounds=int(config["rounds"]),
+                verbose=self.verbose,
+            )
+        else:
+            self.model.set_weights(parameters)
+            history = self.model.fit(
+                self.train_labelled,
+                batch_size=int(config["batch_size"]),
+                epochs=int(config["epochs"]),
+                verbose=self.verbose,
+            )
+        # Print Results
+        print(
+            f"""Client {self.client_id}
+            finished {self.local_train_round}th round of training with
+            loss {history.history['loss'][0]:.4f} and
+            accuracy {history.history['accuracy'][0]:.4f}"""
+        )
+        # Clear Memory
+        # tf.keras.backend.clear_session()
+        gc.collect()
+        return (
+            self.model.get_weights(),
+            self.num_examples_train,

```

Comment:
is it correct reporting only the number of labelled examples?

Reply:
I thought of putting on labeled samples because our semi-supervised architecture is run on the client side and only the unlabelled samples are available there. The server only knows about labeled samples.